### PR TITLE
Changes to GitHub samples to enable them to be directly used by SF Eclipse plugin

### PR DESCRIPTION
**Pull request by sayantancs**
This includes all the changes needed to be done in our github-Java Samples to enable them to be imported  to eclipse and build and deploy them using Service Fabric Eclipse plugin to provide a seamless experience across items we share with outside world.

### CODE REVIEW

File: Actors/ActorCounter/Scripts/upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -x
+azure servicefabric application package copy CounterActorApplication fabric:ImageStore > /dev/null 2>&1
```
Comment (raunakpandya) [inline Actors/ActorCounter/Scripts/upgrade.sh]: So what's the experience now if someone wants to deploy / undeploy using cmdline? I see the install scripts have moved inside the scripts folder. Will they work as the copy command will now look for Application folder inside the script folder. Isnt it?

File: README.md [hunk 2]
```diff
@@ -42,11 +42,15 @@ This stateless service sample demostrates writing a web frontend for your actor
 This stateless service demostrates a watch dog application which is monitoring the EchoServer application. Once you have depployed EchoServer and WatchDog application, you would see that if EchoServer application goes down or for some reason the applications web endpoint is not reachable, an error health is reported by WatchDog for the EchoServer application in the Service Fabric health subsystem. Apart from this, this application also shows inter service communication by forwarding the web requests to Echo Server application. You can view this by going to browser @ http://localhost:12345/getMessage to see the output coming from EchoServer. 
 
 ## Compiling the samples
-For compiling the samples use gradle. This should be configured as part of the Service Fabric Java SDK installation.
+For compilation the samples use gradle. This should be configured as part of the Service Fabric Java SDK installation.
```
Comment (raunakpandya) [inline README.md]: Shouldn't this be compiling only?

File: README.md [hunk 3]
```diff
@@ -42,11 +42,15 @@ This stateless service sample demostrates writing a web frontend for your actor
 This stateless service demostrates a watch dog application which is monitoring the EchoServer application. Once you have depployed EchoServer and WatchDog application, you would see that if EchoServer application goes down or for some reason the applications web endpoint is not reachable, an error health is reported by WatchDog for the EchoServer application in the Service Fabric health subsystem. Apart from this, this application also shows inter service communication by forwarding the web requests to Echo Server application. You can view this by going to browser @ http://localhost:12345/getMessage to see the output coming from EchoServer. 
 
 ## Compiling the samples
-For compiling the samples use gradle. This should be configured as part of the Service Fabric Java SDK installation.
+For compilation the samples use gradle. This should be configured as part of the Service Fabric Java SDK installation.
 
 ## Deploying the samples
 All the samples once compiled can be deployed immediately using the install.sh scripts provided along with the sample. These scripts underneath uses azurecli. Before running the scripts you need to first connect to the cluster using azurecli.
```
Comment (raunakpandya) [inline README.md]: If the install.sh script works from inside scripts folder, we should update here install.sh script which is located in scripts folder in every sample

File: Actors/ActorCounter/Scripts/upgrade.sh (hunk 1 above)
Comment (sayantancs) [inline Actors/ActorCounter/Scripts/upgrade.sh]: yes, it should. we need to run like ``bash Scripts/install.sh`` to do it from terminal at the root/project level. That's how it works internally in eclipse plugin generated SF templates. I just followed the same structure here.

File: README.md (hunk 2 above)
Comment (sayantancs) [inline README.md]: yeah. don't know why I changed it. probably too much reverse doc effect. updated it back to origin.

File: README.md (hunk 3 above)
Comment (sayantancs) [inline README.md]: done.

File: Actors/ActorCounter/Scripts/upgrade.sh (hunk 1 above)
Comment (raunakpandya) [inline Actors/ActorCounter/Scripts/upgrade.sh]: so the script should be fired from root level.. we should capture that in the readme if not already done

File: README.md [hunk 4]
```diff
@@ -45,7 +45,11 @@ This stateless service demostrates a watch dog application which is monitoring t
 For compiling the samples use gradle. This should be configured as part of the Service Fabric Java SDK installation.
 
 ## Deploying the samples
-All the samples once compiled can be deployed immediately using the install.sh scripts provided along with the sample. These scripts underneath uses azurecli. Before running the scripts you need to first connect to the cluster using azurecli. 
+All the samples once compiled can be deployed immediately using the install.sh scripts located inside Scripts folder of every sample. These scripts underneath uses azurecli. Before running the scripts you need to first connect to the cluster using azurecli.
```
Comment (raunakpandya) [inline README.md]: We need to capture the fact that the script must be executed from the root level. If someone goes inside the Scripts folder and then executes install.sh it is likely to fail

File: README.md [hunk 5]
```diff
@@ -45,7 +45,11 @@ This stateless service demostrates a watch dog application which is monitoring t
 For compiling the samples use gradle. This should be configured as part of the Service Fabric Java SDK installation.
 
 ## Deploying the samples
-All the samples once compiled can be deployed immediately using the install.sh scripts provided along with the sample. These scripts underneath uses azurecli. Before running the scripts you need to first connect to the cluster using azurecli. 
+All the samples once compiled can be deployed immediately using the install.sh scripts located inside Scripts folder of every sample. These scripts underneath uses azurecli. Before running the scripts you need to first connect to the cluster using azurecli. 
+
+## Compiling and Deploying the samples using Service Fabric Plugin for Eclipse
+The Service Fabric Plugin for Eclipse can be used to compile and deploy the samples, from Eclipse. To know more about this you can refer to our documentation -
```
Comment (raunakpandya) [inline README.md]: I think its better if we capture eclipse and mac information in this readme itself.

File: README.md (hunk 4 above)
Comment (sayantancs) [inline README.md]: addressed.

File: README.md (hunk 5 above)
Comment (sayantancs) [inline README.md]: addressed.

File: README.md [hunk 6]
```diff
@@ -45,11 +45,28 @@ This stateless service demostrates a watch dog application which is monitoring t
...
+1. Clone the repository in your dev-box - ``https://github.com/Azure-Samples/service-fabric-java-getting-started.git``
+2. On your Eclipse, you need to make sure, Service Fabric Plugin installed.
+3. Create a new workspace.
+4. Open the ``Service Fabric`` Perspective. You can change your Eclipse perspective by following the path - ``Window => Perspective => Open Perspective => Other`` and select  ``Service Fabric`` from the list you see.
+5. To import the GitHub samples in Eclipse, you can go to ``File => Import => General => Existing Project into Workspace`` and select the path to say ``EchoServer1.0``, from your cloned GitHub samples.
+6. Now to build and deploy you can follow the steps mentioned above in the document [Build and deploy the Service Fabric application using Eclipse](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-get-started-eclipse#build-and-deploy-the-service-fabric-application-using-eclipse).
+
+### Import and Deploy Github Java samples on Mac using Service Fabric Eclipse plugin
+The steps mentioned above for Linux, mostly hold true here as well, with minor modifications as mentioned below.
```
Comment (raunakpandya) [inline README.md]: Which steps?

Comment (sayantancs) [inline README.md]: the steps mentioned for linux.. just above..  just above the mac section, there is the linux section.. 
so referring to that should be fine. I guess we don't need to change anything here

**Pull request merged**